src/app/sc2/domain/slack.py

```diff
@@ -0,0 +1,115 @@
+""" Slack integration """
+import urllib
+
+from google.appengine.api import urlfetch
+
+from app.sc2.utils.match_utils import get_vs_string_from_match
+from app.sc2.utils.player_utils import prettify_name
+
+
+# TODO:slack hook this up when auth stuff figured out
+def update_channel_topic_with_open_games(channel_id, open_matches):
+    """
+    Sets the channel topic to the currently open games
+    Need to wait until I check out the auth stuff before this will work
+    """
+    if open_matches:
+        games_to_be_played = '\n'.join([get_vs_string_from_match(m) for m in open_matches])
+    else:
+        games_to_be_played = 'All games have been played! :boom:'
+
+    # TODO: get auth token for slack
+    payload = urllib.urlencode({
+        'token': 'asdf',
+        'channel': channel_id,
+        'purpose': games_to_be_played,
+    })
+
+    headers = {
+        'Content-type': 'application/x-www-form-urlencoded',
+    }
+
+    urlfetch.fetch(
+        'https://slack.com/api/channels.setPurpose',
+        payload=payload,
+        method=urlfetch.POST,
+        headers=headers
+    )
+
+
+def alert_match_closed(match):
+    """
+    Send a message to slack when a match is closed
+    :param match: the match that was closed
+    """
+    headers = {
+        'Content-type': 'application/x-www-form-urlencoded',
+    }
+    winner, loser = determine_match_winner_loser(match)
+    match_string = get_vs_string_from_match(match)
+
+    info_dict = {
+        "fallback": "A match was played/closed",
+        "username": "SC2 Bot",
+    }
+    info_dict.update(get_message_data(winner, loser, match_string))
+
+    payload = urllib.urlencode(info_dict)
+    urlfetch.fetch(
+        'https://hooks.slack.com/services/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',
+        payload=payload,
+        method=urlfetch.POST,
+        headers=headers
+    )
+
+
+def get_message_data(winner, loser, match_string):
+    """
+    Returns a dict of info depending on how the match turned out
+    """
+    winner_names = ' & '.join([prettify_name(n) for n in winner])
+    loser_names = ' & '.join([prettify_name(n) for n in loser])
+    if winner and loser:
+        message = 'Congratulations on the victory {winner}!\nBetter luck next time {loser}'.format(winner=winner_names,
```

Comment:
Lol I almost want to take out the format part here and just have the message say "Better luck next time loser".

Also minor but after `{loser}` it should include some punctuation. Period is fine. They probably don't want an exclamation mark haha.